.github/workflows/json.yml

```diff
@@ -0,0 +1,21 @@
+name: JSON check
+
+on:
+  push:
+    paths:
+      - '**.json'
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: '0'
+      - name: json-check
+        run: |
+          echo "Listing Changed JSON Files"
+          FILES=$(git diff --name-only origin/main HEAD | grep "json")
+          echo "$FILES"
+          echo "$FILES" | xargs -n1 -I {} python -m json.tool {}
```

Comment:
Have you tried different parsers/programmes.
I am asking because I wrote an analyser with serde_json for the whole GSD databse, which finished in ~ 10 min.

The problem with this line is that for every file one python process is executed.

This can be optimised by taking advantage of xarg and dropping the `-n` flag.

```
xarg -0 python -c "$(cat <<EOF
import sys, json
for f in sys.argv[1:]:
  with open(f) as fd:
      json.load(fd)
EOF
)"
```

This should reduces the number of processes by `len(sys.argv[1:])` and the script only writes to stdout in case of an error.

One disadvantage of `xargs` is that it will only find one error.
It would be nicer if all errors could be displayed.